### PR TITLE
Drop EnvironProvider methods

### PR DESCRIPTION
**Pull request by axw**
Drop EnvironProvider.RestrictedConfigAttributes
and EnvironProvider.SecretAttrs methods, and
stop adding credential attributes to and removing
them from model config.

To enable this, there are two functional changes:
 - in the ec2 provider, vpc-id-force is no longer
   restricted. This attribute is only used at
   bootstrap time, so it doesn't need to be the
   same in each model. Preferably it wouldn't be
   in model config at all, but we don't have a
   way of describing provider-specific bootstrap
   config.
 - in the manual provider, we drop the bootstrap-user
   config attribute and extend the endpoint syntax
   to permit user@host as well as host.

### CODE REVIEW

**Comment by axw** [comment]
**QA**

1. bootstrap azure
2. add-model foo
3. get-model-config (check no credentials included)
4. check controller logs, make sure there are no warnings about unknown config attributes

**Comment by wallyworld** [comment]
lgtm

File: provider/manual/environ.go [hunk 1]
```diff
@@ -272,9 +275,34 @@ var unsupportedConstraints = []string{
 func (e *manualEnviron) ConstraintsValidator() (constraints.Validator, error) {
 	validator := constraints.NewValidator()
 	validator.RegisterUnsupported(unsupportedConstraints)
+	if isRunningController() {
```
Comment (anastasiamac) [inline provider/manual/environ.go]: Nice !

File: provider/manual/environ.go [hunk 2]
```diff
@@ -272,9 +275,34 @@ var unsupportedConstraints = []string{
...
+		validator.UpdateVocabulary(constraints.Arch, []string{arch.HostArch()})
+	} else {
+		// We're running outsode of the Juju controller, so we must
```
Comment (anastasiamac) [inline provider/manual/environ.go]: typo

**Comment by axw** [comment]
$$merge$$

**Comment by axw** [comment]
This fixed a critical issue with the manual provider arch constraints, JFDIing.

$$__jfdi__$$

**Comment by axw** [comment]
$$__JFDI__$$

**Pull request merged**